felix/daemon/daemon.go

```diff
@@ -591,15 +591,9 @@ configRetry:
 			break
 		}
 		healthAggregator.Report(healthName, &health.HealthReport{Live: true, Ready: true})
-
-		supportsNodeResourceUpdates, err := typhaConnection.SupportsNodeResourceUpdates(10 * time.Second)
-		if err != nil {
-			time.Sleep(time.Second) // Avoid tight restart loop in case we didn't really wait 10s above.
-			log.WithError(err).Fatal("Did not get hello message from Typha in time")
-			return
-		}
-		log.Debugf("Typha supports node resource updates: %v", supportsNodeResourceUpdates)
-		configParams.SetUseNodeResourceUpdates(supportsNodeResourceUpdates)
+		// Up-to-date Typha client will refuse to connect unless Typha signals
+		// that it supports node resource updates.
```

Comment:
Any issue with version skew on-upgrade?

Reply:
I think the feature was added in 2019 so, yes, but only if you're skipping a dozen versions.  Even in that case, it will only block new felix from talking to old typha, felix will just keep trying untill it connects to an up-level typha.